### PR TITLE
Add modulus operator support

### PR DESCRIPTION
**Pull request by MingchenZhang**
Programing feature is very useful in de-multiplexing multiplexed channel. Multiplexed channel could help increase number of switches on a receiver with limited channel number. 
Adding a modulus operator makes the de-multiplexing much more straightforward. This could save one operator comparing to doing a division, multiplication and subtraction to get next significant "digit".

### CODE REVIEW

**Comment by MingchenZhang** [comment]
https://github.com/iNavFlight/inav-configurator/pull/1196 is the configurator side pull request.

File: src/main/programming/logic_condition.h [hunk 1]
```diff
@@ -47,29 +47,30 @@ typedef enum {
...
-    LOGIC_CONDITION_SET_OSD_LAYOUT              = 32,
-    LOGIC_CONDITION_SIN                         = 33,
-    LOGIC_CONDITION_COS                         = 34,
-    LOGIC_CONDITION_TAN                         = 35,
-    LOGIC_CONDITION_MAP_INPUT                   = 36,
-    LOGIC_CONDITION_MAP_OUTPUT                  = 37,
-    LOGIC_CONDITION_RC_CHANNEL_OVERRIDE         = 38,
-    LOGIC_CONDITION_SET_HEADING_TARGET          = 39,
-    LOGIC_CONDITION_LAST                        = 40,
+    LOGIC_CONDITION_MODULUS                     = 18,
```
Comment (DzikuVx) [inline src/main/programming/logic_condition.h]: Please add it a LAST entry not to break compatibility with existing solutions

**Review by DzikuVx — CHANGES_REQUESTED** [review]

File: src/main/programming/logic_condition.h (hunk 1 above)
Comment (Tygrys-1) [reply src/main/programming/logic_condition.h]: I think that also numbers of conditions should be preserved, so the Modulus is number 40, and the LAST goes to 41.

Comment (MingchenZhang) [reply src/main/programming/logic_condition.h]: I moved it to the last entry, one of the commit i referred to moved the enum number around so i just followed suit.

**Review by DzikuVx — APPROVED** [review]

**Pull request merged**